### ISSUE TITLE
 Unhandled exception: Premature end of stream

### ISSUE THREAD

**Issue by DestruidorPT** [abergs/fido2-net-lib#216]
```
crit: IdentityServer4.Hosting.IdentityServerMiddleware[0]
      Unhandled exception: Premature end of stream
PeterO.Cbor.CBORException: Premature end of stream
   at PeterO.Cbor.CBORReader.ReadByteData(Stream stream, Int64 uadditional, Stream outputStream)
   at PeterO.Cbor.CBORReader.ReadStringArrayMap(Int32 type, Int64 uadditional)
   at PeterO.Cbor.CBORReader.ReadInternal()
   at PeterO.Cbor.CBORReader.ReadStringArrayMap(Int32 type, Int64 uadditional)
   at PeterO.Cbor.CBORReader.ReadInternal()
   at PeterO.Cbor.CBORReader.Read()
   at PeterO.Cbor.CBORObject.Read(Stream stream, CBOREncodeOptions options)
   at PeterO.Cbor.CBORObject.DecodeFromBytes(Byte[] data, CBOREncodeOptions options)
   at PeterO.Cbor.CBORObject.DecodeFromBytes(Byte[] data)
   at Fido2NetLib.Objects.CredentialPublicKey..ctor(Byte[] cpk)
   at Fido2NetLib.AuthenticatorAssertionResponse.VerifyAsync(AssertionOptions options, String expectedOrigin, Byte[] storedPublicKey, UInt32 storedSignatureCounter, IsUserHandleOwnerOfCredentialIdAsync isUserHandleOwnerOfCredId, Byte[] requestTokenBindingId)
   at Fido2NetLib.Fido2.MakeAssertionAsync(AuthenticatorAssertionRawResponse assertionResponse, AssertionOptions originalOptions, Byte[] storedPublicKey, UInt32 storedSignatureCounter, IsUserHandleOwnerOfCredentialIdAsync isUserHandleOwnerOfCredentialIdCallback, Byte[] requestTokenBindingId)

fail: Microsoft.AspNetCore.Server.Kestrel[13]
      Connection id "0HM844518RFR3", Request id "0HM844518RFR3:00000001": An unhandled exception was thrown by the application.
PeterO.Cbor.CBORException: Premature end of stream
   at PeterO.Cbor.CBORReader.ReadByteData(Stream stream, Int64 uadditional, Stream outputStream)
   at PeterO.Cbor.CBORReader.ReadStringArrayMap(Int32 type, Int64 uadditional)
   at PeterO.Cbor.CBORReader.ReadInternal()
   at PeterO.Cbor.CBORReader.ReadStringArrayMap(Int32 type, Int64 uadditional)
   at PeterO.Cbor.CBORReader.ReadInternal()
   at PeterO.Cbor.CBORReader.Read()
   at PeterO.Cbor.CBORObject.Read(Stream stream, CBOREncodeOptions options)
   at PeterO.Cbor.CBORObject.DecodeFromBytes(Byte[] data, CBOREncodeOptions options)
   at PeterO.Cbor.CBORObject.DecodeFromBytes(Byte[] data)
   at Fido2NetLib.Objects.CredentialPublicKey..ctor(Byte[] cpk)
   at Fido2NetLib.AuthenticatorAssertionResponse.VerifyAsync(AssertionOptions options, String expectedOrigin, Byte[] storedPublicKey, UInt32 storedSignatureCounter, IsUserHandleOwnerOfCredentialIdAsync isUserHandleOwnerOfCredId, Byte[] requestTokenBindingId)
   at Fido2NetLib.Fido2.MakeAssertionAsync(AuthenticatorAssertionRawResponse assertionResponse, AssertionOptions originalOptions, Byte[] storedPublicKey, UInt32 storedSignatureCounter, IsUserHandleOwnerOfCredentialIdAsync isUserHandleOwnerOfCredentialIdCallback, Byte[] requestTokenBindingId)
```

I am getting this error only when I authenticate using Platform with Windows Hello PIN. 
When I use Platform with Android browser works without any problem and in Cross-platform with Yubikey also works, only the Platform with Windows Hello PIN give me that error.

Platform with Windows Hello PIN:
**Assertion Options:**
```
{
    "challenge": "0Suv6zwHC9f_ZPsp2yH1-Q",
    "timeout": 60000,
    "rpId": "exemple.com",
    "allowCredentials": [
        {
            "type": "public-key",
            "id": "joXfhGtqNMdEYt_C9vuqlZ1xXpujTTw3sY2w3-SucBE",
            "transports": [
                "internal"
            ]
        },
        {
            "type": "public-key",
            "id": "d6cUunNV6WWOw_2ruO7uTBLxnbt9Ek-x1OwFL1fkjYH9ecMfuhxmDT75uSrZqUZizDmfE7iofv_1ZeVNg4IgbQ",
            "transports": [
                "ble",
                "nfc",
                "usb"
            ]
        }
    ],
    "userVerification": "discouraged",
    "status": "ok",
    "errorMessage": ""
}
```
**Authenticator Attestation:**
```
{
    "id":"joXfhGtqNMdEYt_C9vuqlZ1xXpujTTw3sY2w3-SucBE",
    "rawId":"joXfhGtqNMdEYt_C9vuqlZ1xXpujTTw3sY2w3-SucBE",
    "type":"public-key",
    "response":{
            "clientDataJSON":      "eyJ0eXBlIjoid2ViYXV0aG4uZ2V0IiwiY2hhbGxlbmdlIjoiMFN1djZ6d0hDOWZfWlBzcDJ5SDEtUSIsIm9yaWdpbiI6Imh0dHBzOi8vZWx0b25wYXN0aWxoYS5jb20iLCJjcm9zc09yaWdpbiI6ZmFsc2UsIm90aGVyX2tleXNfY2FuX2JlX2FkZGVkX2hlcmUiOiJkbyBub3QgY29tcGFyZSBjbGllbnREYXRhSlNPTiBhZ2FpbnN0IGEgdGVtcGxhdGUuIFNlZSBodHRwczovL2dvby5nbC95YWJQZXgifQ",
            "authenticatorData": "JODq-RrrOA6E2bL8N35CWgjO8ZWu54VrEh_dsgXH_uQFAAAAEA",
            "signature": "WsyTo_vKVe1G5lGDYtZlLA1vF0cr6Ir6STX8XyQNl_NeJZ6-ILCWGvzaWT__mNO3X_CssWtkqjvLvV3NxK6T_U_WsIRYk-SAD-igROXqGe9Yv8LlUuOoH9N2MoKZGnFa5oPpN0IWoWdbKmyLZPCIfS8Sqhxw6U8Z4NeBILBNAPJCUvDo6hCb87Xdi7MgN3Bm_h0Lhe8qZhj-m2nJQ4u0HdVHvp2kkrGZgTAdPPXBaiobs5NDF_7CbafUYyGoy0naPPjwL9i7pjOm931V8rjcIrWpiz2iz5LsD8mfp7pQWMbY5zks5Yfxm1TctF_opHbLRPTFfD7BCEuI0qVWVlf0Eg",
            "userHandle": "QfY7MV5-ek-k46zHAVZjvw"
    },
    "extensions":{}
}
```

Does anyone know what is happening? From what I test maybe the clientDataJSON is giving problems.

**Comment by aseigler** [comment]
The thing you have labeled "Authenticator Attestation" is actually an assertion, and it [looks pretty reasonable](https://debugger.simplewebauthn.dev/?assertion=ewogICAgImlkIjoiam9YZmhHdHFOTWRFWXRfQzl2dXFsWjF4WHB1alRUdzNzWTJ3My1TdWNCRSIsCiAgICAicmF3SWQiOiJqb1hmaEd0cU5NZEVZdF9DOXZ1cWxaMXhYcHVqVFR3M3NZMnczLVN1Y0JFIiwKICAgICJ0eXBlIjoicHVibGljLWtleSIsCiAgICAicmVzcG9uc2UiOnsKICAgICAgICAgICAgImNsaWVudERhdGFKU09OIjogICAgICAgICAgICAgImV5SjBlWEJsSWpvaWQyVmlZWFYwYUc0dVoyVjBJaXdpWTJoaGJHeGxibWRsSWpvaU1GTjFkalo2ZDBoRE9XWmZXbEJ6Y0RKNVNERXRVU0lzSW05eWFXZHBiaUk2SW1oMGRIQnpPaTh2Wld4MGIyNXdZWE4wYVd4b1lTNWpiMjBpTENKamNtOXpjMDl5YVdkcGJpSTZabUZzYzJVc0ltOTBhR1Z5WDJ0bGVYTmZZMkZ1WDJKbFgyRmtaR1ZrWDJobGNtVWlPaUprYnlCdWIzUWdZMjl0Y0dGeVpTQmpiR2xsYm5SRVlYUmhTbE5QVGlCaFoyRnBibk4wSUdFZ2RHVnRjR3hoZEdVdUlGTmxaU0JvZEhSd2N6b3ZMMmR2Ynk1bmJDOTVZV0pRWlhnaWZRIiwKICAgICAgICAgICAgImF1dGhlbnRpY2F0b3JEYXRhIjogIkpPRHEtUnJyT0E2RTJiTDhOMzVDV2dqTzhaV3U1NFZyRWhfZHNnWEhfdVFGQUFBQUVBIiwKICAgICAgICAgICAgInNpZ25hdHVyZSI6ICJXc3lUb192S1ZlMUc1bEdEWXRabExBMXZGMGNyNklyNlNUWDhYeVFObF9OZUpaNi1JTENXR3Z6YVdUX19tTk8zWF9Dc3NXdGtxanZMdlYzTnhLNlRfVV9Xc0lSWWstU0FELWlnUk9YcUdlOVl2OExsVXVPb0g5TjJNb0taR25GYTVvUHBOMElXb1dkYktteUxaUENJZlM4U3FoeHc2VThaNE5lQklMQk5BUEpDVXZEbzZoQ2I4N1hkaTdNZ04zQm1faDBMaGU4cVpoai1tMm5KUTR1MEhkVkh2cDJra3JHWmdUQWRQUFhCYWlvYnM1TkRGXzdDYmFmVVl5R295MG5hUFBqd0w5aTdwak9tOTMxVjhyamNJcldwaXoyaXo1THNEOG1mcDdwUVdNYlk1emtzNVlmeG0xVGN0Rl9vcEhiTFJQVEZmRDdCQ0V1STBxVldWbGYwRWciLAogICAgICAgICAgICAidXNlckhhbmRsZSI6ICJRZlk3TVY1LWVrLWs0NnpIQVZaanZ3IgogICAgfSwKICAgICJleHRlbnNpb25zIjp7fQp9Cgo).  I am pretty the issue is going to be the stored public key, it is checked and [loaded here](https://github.com/abergs/fido2-net-lib/blob/83b82143361ae0c250a7e15a74b50da1866c08c6/Src/Fido2/AuthenticatorAssertionResponse.cs#L162).  It must be corrupted to the point where it is no longer valid CBOR.  How are you storing the public keys?  In the attestation verifier, the exact same check is done on the public key before it is handed back to the caller for storage, so it had to get stomped on after that.

**Comment by DestruidorPT** [comment]
@aseigler Thx, it was the length of public-key on my database was too small.
I have another problem, is about the origin that comes from the Android, here is error:

 ```
Unhandled exception: Origin android:apk-key-hash:LBpmHljqwJJLdXVNpjfdAz2k2MNojcSFZZIuRj1B4wI not equal to original origin android:apk-key-hash:LBpmHljqwJJLdXVNpjfdAz2k2MNojcSFZZIuRj1B4wI
Fido2NetLib.Fido2VerificationException: Origin android:apk-key-hash:LBpmHljqwJJLdXVNpjfdAz2k2MNojcSFZZIuRj1B4wI not equal to original origin android:apk-key-hash:LBpmHljqwJJLdXVNpjfdAz2k2MNojcSFZZIuRj1B4wI
   at Fido2NetLib.AuthenticatorResponse.BaseVerify(String expectedOrigin, Byte[] originalChallenge, Byte[] requestTokenBindingId)
   at Fido2NetLib.AuthenticatorAssertionResponse.VerifyAsync(AssertionOptions options, String expectedOrigin, Byte[] storedPublicKey, UInt32 storedSignatureCounter, IsUserHandleOwnerOfCredentialIdAsync isUserHandleOwnerOfCredId, Byte[] requestTokenBindingId)
   at Fido2NetLib.Fido2.MakeAssertionAsync(AuthenticatorAssertionRawResponse assertionResponse, AssertionOptions originalOptions, Byte[] storedPublicKey, UInt32 storedSignatureCounter, IsUserHandleOwnerOfCredentialIdAsync isUserHandleOwnerOfCredentialIdCallback, Byte[] requestTokenBindingId)
```

This happens because android is sending this origin "android:apk-key-hash:LBpmHljqwJJLdXVNpjfdAz2k2MNojcSFZZIuRj1B4wI" that is correct but in [AuthenticatorResponse](https://github.com/abergs/fido2-net-lib/blob/master/Src/Fido2/AuthenticatorResponse.cs) you guys have in BaseVerify on line 74 this:

```
            if (!string.Equals(FullyQualifiedOrigin(this.Origin), expectedOrigin, StringComparison.OrdinalIgnoreCase))
                throw new Fido2VerificationException($"Origin {Origin} not equal to original origin {expectedOrigin}");
```

The function "FullyQualifiedOrigin" cause the origin "android:apk-key-hash:LBpmHljqwJJLdXVNpjfdAz2k2MNojcSFZZIuRj1B4wI"   to become this "android://", there is any temporary solution?

**Comment by aseigler** [comment]
Yes, I fixed that today via #217.  Expect a v2.0.1 release soon with that fix.

**Comment by abergs** [comment]
2.0.1 is out! https://www.nuget.org/packages/Fido2
